GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs

```diff
@@ -354,10 +350,9 @@ private void removeRecentToolStripMenuItem_Click(object sender, EventArgs e)
                 {
                     _repositoryHistory = await RepositoryHistoryManager.Locals.RemoveRecentAsync(repo.Repo.Path);
                 }
-
-                await this.SwitchToMainThreadAsync();
-                RefreshRepos();
             });
+            ThreadHelper.AssertOnUIThread();
```

Comment:
💭 This one _could_ move to the beginning of the method, but click handlers seem obviously on the main thread so may be redundant?